### ISSUE TITLE
Problem got when trying to build OpenSfM on centos

### ISSUE THREAD

**Issue by allenChn** [mapillary/OpenSfM#378]
Hello, dear friends in mapillary.
I got a problem when I tried to build OpenSfM on CentOS. This is the screenshot of the problem.
![image](https://user-images.githubusercontent.com/20311590/48465754-da10d800-e81e-11e8-859e-b259c1bbe748.png)
The compiling env is like this:

- centos 6

- python2.7

- gcc 6.4.0

- cmake 3.3.2

- other dependencies referring to the Dockerfile given in OpenSfM

It's not over..............................................................................................................
I didn't give up.
I tried another solution. I ran 'docker build' using Dockerfile.python3 on a server of centos7(with docker and python3 and pip3 installed). However, the problem still exists. I have no idea whether I am still misssing some dependencies, or I  just can't use centos to build.
Since We use CentOS(rather than Ubuntu) in most of our cluster servers, we have to build opensfm with commands like 'yum' and searching for the replacing requirements in Dockerfile one by one. It's complicated to verify whether I install the right one. Hoping for your solution! Thank you.

**Comment by paulinus** [comment]
Hi @allenChn,
We've not tried installing on CentOS, so thanks for reporting and helping getting it running.
The error message complains about missing Pybind11 commands. We are getting pybind11 as a git submodule. You make sure that you get the submodules updated by running
```
git submodule update --init --recursive
```
If not done already, could you please try this command and check if the error remain?
thanks

**Comment by allenChn** [comment]
I will try that. Thank you very much.

**Comment by GITNE** [comment]
Looks like your GCC's C++ cmake module may be lacking some checks which should populate either `PYBIND11_CPP_STANDARD` or `CMAKE_CXX_STANDARD` variables.

You can try running
```sh
cmake -D PYBIND11_CPP_STANDARD='-std=c++11'
```
or you can modify your `CMakeCache.txt` file and set `PYBIND11_CPP_STANDARD` to `-std=c++11`.

Alternatively, you can try
```sh
cmake -D HAS_CPP11_FLAG
```
and pybind will set the `PYBIND11_CPP_STANDARD` variable for you.

**Comment by allenChn** [comment]
Thanks a lot for your suggestion. If I succeed in building OpenSfM on centos, I will contribute my dockerfile.

**Comment by allenChn** [comment]
Actually I succeeded in building OpenSfM on a new centos7 server with your 'Dockerfile.python3' provided in project. I ran berlin example and no problem arose. It seems that there's no need to build opensfm on centos all by myself, since I can use opensfm in a docker container. Is that right?

**Comment by paulinus** [comment]
That is right. If you use the docker container, then there's no need to build the lib locally.
